components/RequestData.js

```diff
@@ -0,0 +1,37 @@
+const noflo = require('noflo');
+
+exports.getComponent = function () {
+  const c = new noflo.Component();
+  c.description = 'Request data from source for a HTTP request';
+  c.icon = 'share-square-o';
+  c.inPorts.add('req', {
+    datatype: 'object',
+    description: 'Incoming HTTP request',
+    scoped: true,
+  });
+  c.outPorts.add('req', {
+    datatype: 'object',
+    description: 'HTTP request',
+    scoped: true,
+  });
+  c.outPorts.add('payload', {
+    datatype: 'object',
+    description: 'Request payload',
+    scoped: true,
+    addressable: true,
```

Comment:
Why addressable? What is the meaning/use of the different connection indices?

Reply:
When you have multiple offerings, this allows routing the data request to the correct data provider (same index as the offering was registered with).

For example:

```fbp
'offering1' -> OFFERING[0] Provider
'offering2' -> OFFERING[1] Provider
Provider REQUEST[0] -> GetOffering1Data -> RESPONSE Provider
Provider REQUEST[1] -> GetOffering2Data -> RESPONSE Provider
```

Reply:
Note: the `RESPONSE` port is not indexed, since we already match the correct data to correct request via NoFlo scopes.